samples/append_rows_proto2.js

```diff
@@ -0,0 +1,313 @@
+      row.row_num = 3;
+      row.setBytesCol(Buffer.from('later, gator'));
+      serializedRows.push(row.serializeBinary());
+
+      // Row 4
+      row = new sample_data_pb2.SampleData();
+      row.row_num = 4;
+      row.setFloat64Col(987.654);
+      serializedRows.push(row.serializeBinary());
+
+      // Row 5
+      row = new sample_data_pb2.SampleData();
+      row.row_num = 5;
+      row.setInt64Col(321);
+      serializedRows.push(row.serializeBinary());
+
+      // Row 6
+      row = new sample_data_pb2.SampleData();
+      row.row_num = 6;
+      row.setStringCol('octavia');
+      serializedRows.push(row.serializeBinary());
+
+      let protoRows = {
+        writerSchema: {protoDescriptor},
+        rows: {serializedRows},
+      };
+
+      // Construct request.
+      request = {
+        writeStream,
+        protoRows,
+      };
+
+      // Send batch.
+      stream.write(request);
+
+      // Reset rows.
+      serializedRows = [];
+
+      // Row 7
+      row = new sample_data_pb2.SampleData();
+      row.row_num = 7;
+      row.setDateCol(1132896);
+      serializedRows.push(row.serializeBinary());
+
+      // Row 8
+      row = new sample_data_pb2.SampleData();
+      row.row_num = 8;
+      row.setDatetimeCol('2019-02-17 11:24:00.000');
+      serializedRows.push(row.serializeBinary());
+
+      // Row 9
+      row = new sample_data_pb2.SampleData();
+      row.row_num = 9;
+      row.setGeographyCol('POINT(5 5)');
+      serializedRows.push(row.serializeBinary());
+
+      // Row 10
+      row = new sample_data_pb2.SampleData();
+      row.row_num = 10;
+      row.setNumericCol('123456');
+      row.setBignumericCol('99999999999999999999999999999.999999999');
+      serializedRows.push(row.serializeBinary());
+
+      // Row 11
+      row = new sample_data_pb2.SampleData();
+      row.row_num = 11;
+      row.setTimeCol('18:00:00');
+      serializedRows.push(row.serializeBinary());
+
+      // Row 12
+      row = new sample_data_pb2.SampleData();
+      row.row_num = 12;
+      const timestamp = Date.now();
+      row.setTimestampCol(timestamp);
+      serializedRows.push(row.serializeBinary());
+
+      // Since this is the second request, you only need to include the row data.
+      // The name of the stream and protocol buffers DESCRIPTOR is only needed in
+      // the first request.
+      protoRows = {
+        rows: {serializedRows},
+      };
+
+      request = {
+        protoRows,
+      };
+
+      // Send batch.
+      stream.write(request);
+
+      serializedRows = [];
+
+      // Row 13
+      row = new sample_data_pb2.SampleData();
+      row.row_num = 13;
+      row.addInt64List(1999);
+      row.addInt64List(2001);
+      serializedRows.push(row.serializeBinary());
+
+      protoRows = {
+        rows: {serializedRows},
+      };
+
+      request = {
+        protoRows,
+      };
+
+      // Send batch.
+      stream.write(request);
+
+      stream.end();
```

Comment:
I think you'll want to wait until you get all the expected responses back before ending the stream. See my comment above about PENDING mode.